### DISCUSSION TITLE
2021-07-13 Node Experience Team Update

### DISCUSSION THREAD

**Discussion by janewang** [near/community#73] [Node]
# Brief Summary

This past week, the Node Experience team focused on improving pytests and nayduck tests, adding features to show better metrics and node status, fuzzing the runtime, and setting up grants to support roadmap items. The team has done some housekeeping, including reviewing `T-nodeX` issues and sorting the ones to focus on into our Q3 board, and reviewing the current weekly process on the team. 

---

Node Experience team project board: https://github.com/orgs/near/projects/18
Attended: @mina86 @nikurt @posvyatokum @bowenwang1996 @janewang 

# Done
- @mina86 completed the issue to remove pytest's dependence on the NEAR SDK and on the master branch throughout the repository. Removing the dependency also sped up the tests. near/nearcore#4480
- @janewang improved the docs with better description for localnet, betanet, testnet, and mainnet near/docs#777
- @janewang and @bowenwang1996 reviewed the `T-nodeX` issues and sorted these issues into the right status into NodeX Q3 board
- @bowenwang1996 fixed some Nayduck expensive tests https://github.com/near/nearcore/pull/4500

# In Progress
- @mina86 continues to work on Nayduck transition and had gained access to the machines. He also fixed security issues with Nayduck. near/nayduck#1
- @mina86 is working on improving logging for failed tests and understand which command caused the failure for Nayduck. near/nayduck#1
- @nikurt is working on exposing Prometheus metrics on a custom port apart from the RPC port, a feature requested from one of our validators. near/nearcore#2693
- @nikhurt is working on reporting the earliest block hash, height, and time in node status response for non-archival nodes since old blocks are gc'ed. near/nearcore#3750
- @posvyatokum submitted a skeleton for runtime tests. near/nearcore#4335
- @posvyatokum has implemented the function for fuzzing to work with arbitrary scenarios and can publish the basic design without the fuzzing target. near/nearcore#4374
- @janewang worked on grants for Open Shards Alliance, Guildnet, RPC, and BisonTrails and will transition these on Monday
- @janewang is working to create processes in NVAB

# Blocked
- @mina86 is blocked on adding CI test that checks that attempt publishing crates and compile them to Wasm. A decision is to be made on approach and the story is waiting for response from tagged stakeholders. near/nearcore#4379. 
- @posvyatokum has implemented function for fuzzing to work with arbitrary scenarios but may be blocked by Nightly and has an issue with compilation, which runs out of memory. near/nearcore#4374

# Meta
- The team feels the weekly team call is a good place to connect and there's the right amount of process. 👍 

# Upcoming
- NodeX Game Hour this Friday with Geoguessr 🎈 
- Mainnet release of 1.20.0 is on Monday, July 26th ⚙️

**Comment by maxzaver** [comment]
Seems like https://github.com/near/nearcore/issues/4379 is unblocked. @janewang can you confirm?

**Reply by janewang** [reply]
Confirmed, near/nearcore#4379 is unblocked. near/nearcore#4374 is also unblocked 👍
